src/data/polynomial/erase_lead.lean

```diff
@@ -159,4 +159,30 @@ end
 
 end erase_lead
 
+/--  An induction lemma for polynomials. It takes a natural number `N` as a parameter, that is
+required to be at least as big as the `nat_degree` of the polynomial.  This is useful to prove
+results where you want to change each term in a polynomial to something else depending on the
+`nat_degree` of the polynomial itself and not on the specific `nat_degree` of each term. -/
+lemma induction_with_nat_degree_le {R : Type*} [semiring R] {P : polynomial R → Prop} (N : ℕ)
```

Comment:
```suggestion
@[elab_as_eliminator]
lemma induction_with_nat_degree_le {R : Type*} [semiring R] {P : polynomial R → Prop} (N : ℕ)
```
I confess that I don't know the details, but this attribute I put on all induction principles. I guess this is a good moment for me to look up what it does exactly.

Reply:
I accepted this, although I am sure that I know less than you what `elab_as_eliminator` does!

Reply:
When applying the induction principle, I now get an error:
```
type mismatch at application
  induction_with_nat_degree_le N (is_integer_eval_zero N a b)
term
  is_integer_eval_zero N a b
has type
  ∃ (D : R), ⇑?m_3 D = ⇑?m_3 b ^ N * eval (⇑?m_3 a * ?m_4) (map ?m_3 0)
but is expected to have type
  ?m_1 0
Additional information:
/home/damiano/Matematica/Lean/mathlib_PRs/src/data/polynomial/denominators.lean:52:9: context: 'eliminator' elaboration was not used for 'induction_with_nat_degree_le' because it is not fully applied, #6 explicit arguments expected
```
If I use `refine N _ _ _`, then the command correctly produces three goals that I can solve with 3 `exact`s.  However, if I fill in myself the underscores with the relevant terms, I get the error above (or something similar, depending on which blank I try to fill in).  I would either revert to the "not `elab as eliminator` version, or try to understand the `elab`.  Do you know where to look for this?

Thanks!

Reply:
Found an explanation: https://leanprover.github.io/theorem_proving_in_lean/interacting_with_lean.html#elaboration-hints